code/clue.js

```diff
@@ -1,59 +1,396 @@
+const shuffleWeapons = () => {
+  suspects.forEach(suspect => {
+    suspect.favorite = randomSelector(weapons).name
+  })
+}
+
+//Function to increase probability of favourite weapon to be assigned to killer:
+//THIS FUNCTION DOESN'T WORK???
+const increaseChanceOfFavorite = () => {
+  const favWeapon = weapons.find(weapon => weapon.id === mystery.killer.favorite);
+  const favWeaponArray = new Array(7).fill(favWeapon);
+  console.log(favWeaponArray)
+  const calculatedWeapon = [...weapons, ...favWeaponArray];
+  return calculatedWeapon;
+};
 
 // This function will be invoked when you click on the killer card.
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
-
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  if (!killerIsPicked) {
+    document.getElementById('killerLoader').style.visibility='visible' 
+    setTimeout(function () {
+      document.getElementById('killerLoader').style.visibility='hidden'
+      //excecutes shuffleWeapons-function above and makes favorite weapon random:
+      shuffleWeapons();
+      //randomly select a killer from the suspects. And add that to the mystery object.
+      mystery.killer = randomSelector(suspects); //mystery.killer selects the property 'killer' in mystery object and sets this equal to a randomly selected suspect. 
+      document.getElementById('killerCard').style.background = mystery.killer.color
+      document.getElementById('killerName').innerHTML = `Name: ${mystery.killer.firstName} ${mystery.killer.lastName}`
+      document.getElementById('killerAge').innerHTML = `Age: ${mystery.killer.age}`
+      document.getElementById('killerOccupation').innerHTML = `Occupation: ${mystery.killer.occupation}`
+      document.getElementById('killerWeapon').innerHTML = `Favorite weapon: ${mystery.killer.favorite}`
+      document.getElementById('killerDescription').innerHTML = `Description: ${mystery.killer.description}`
+      document.getElementById('killerImage').src = `${mystery.killer.image}`
+      killerIsPicked = true;
+    }, 2500)
+  }
+};
+//Added eventlistener below instead of "onclick" in HTML. Don't know which way is the best way to do it. This only works if the eventlistener is outside of the function. 
+//First param: type of event. Second param: the function to call on event.
+document.getElementById('killerCard').addEventListener('click', pickKiller);
+
+//If pickKiller function is excecuted/card is clicked then excecute this function:
+const pickWeapon = () => {
+  if (killerIsPicked && !weaponIsPicked) {
```

Comment:
I like that you can only select a card once the previous animation has finished.

Reply:
Indeed, nice touch!